Sources/XCTest/XCTAssert.swift

```diff
@@ -87,8 +87,12 @@ private func _XCTEvaluateAssertion(assertion: _XCTAssertion, @autoclosure messag
     case .Success:
         return
     default:
-        if let handler = XCTFailureHandler {
-            handler(XCTFailure(message: message(), failureDescription: result.failureDescription(assertion), expected: result.expected, file: file, line: line))
+        if let currentTestCase = XCTCurrentTestCase {
```

Comment:
test_hasInitializedValue' passed \(\d+\.\d+ seconds\).
func test_hasInitializedValue() {
XCTAssertEqual(value, 1)
value += 1
}
-// CHECK: Test Case
'NewInstanceForEachTestTestCase.test_hasInitializedValueInAnotherTest'
started.
+// CHECK: Test Case
'NewInstanceForEachTestTestCase.test_hasInitializedValueInAnotherTest'
started at \d+:\d+:\d+\.\d+
// CHECK: Test Case
'NewInstanceForEachTestTestCase.test_hasInitializedValueInAnotherTest'
passed \(\d+\.\d+ seconds\).
func test_hasInitializedValueInAnotherTest() {
XCTAssertEqual(value, 1)
}
}
-// CHECK: Executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+
\(\d+\.\d+\) seconds
+// CHECK: Test Suite 'NewInstanceForEachTestTestCase' passed at
\d+:\d+:\d+\.\d+
+// CHECK: \t Executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+

2016-04-03 21:13 GMT+02:00 Brian Croom <notifications@github.com>:

> In Sources/XCTest/XCTAssert.swift
> <https://github.com/apple/swift-corelibs-xctest/pull/86#discussion_r58315746>
> :
>
> > @@ -87,8 +87,12 @@ private func _XCTEvaluateAssertion(assertion: _XCTAssertion, @autoclosure messag
> >      case .Success:
> >          return
> >      default:
> > -        if let handler = XCTFailureHandler {
> > -            handler(XCTFailure(message: message(), failureDescription: result.failureDescription(assertion), expected: result.expected, file: file, line: line))
> > +        if let currentTestCase = XCTCurrentTestCase {
>
> [image: :+1:]
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/apple/swift-corelibs-xctest/pull/86/files/651fd407f55f25eba400bf19003553f236e22083#r58315746>
>